### PR TITLE
Fix use-after-free bug in peephole pass

### PR DESCRIPTION
**Pull request by bddppq**
@ezyang @apaszke

### CODE REVIEW

File: torch/csrc/jit/passes/peephole.cpp [hunk 1]
```diff
@@ -39,12 +41,15 @@ void PeepholeOptimize(Block * block) {
...
         }
         break;
     }
 
-    for (Block * sub_block : n->blocks()) {
-      PeepholeOptimize(sub_block);
+    if (!destroyed) {
+        for (Block * sub_block : n->blocks()) {
+            PeepholeOptimize(sub_block);
+        }
```
Comment (apaszke) [inline torch/csrc/jit/passes/peephole.cpp]: Couldn’t we just move this loop before the switch? The order doesn’t really matter here, and that’s less bookkeeping

Comment (bddppq) [reply torch/csrc/jit/passes/peephole.cpp]: @apaszke ok done

**Review by apaszke — APPROVED** [review]

**Comment by ezyang** [comment]
Hi:

1. Did any of our current tests catch this bug?
2. If so, why didn't the ASAN build catch the bug?

Thanks.

**Review by ezyang — CHANGES_REQUESTED** [review]
Want to know if a test is necessary

**Comment by bddppq** [comment]
@ezyang This is very likely covered by tests in onnx-fb-universe, however its CI is not running in ASAN mode. Considering the those tests are going to be moved soon, probably not worth beefing up its CI setup?

**Comment by ezyang** [comment]
I wouldn't suggest updating onnx-fb-universe, but I am wondering why the peephole optimizer (it's NOT onnx) isn't being sufficiently exercised in `test_jit.py`. How did you discover the bug?

**Comment by bddppq** [comment]
Yeah I know it's optimizing the trace. I'm saying there are plenty of export tests in onnx-fb-universe and they are going to be moved and after the move there will be asan build of them.
We were exporting a model in fbcode and it builds in asan mode.

**Review by ezyang — APPROVED** [review]

**Pull request merged**